### DISCUSSION TITLE
establish communication between onWSM() of MyVeinsApp.cc and RSUApp.cc

### DISCUSSION THREAD

**Discussion by gugo8704** [riebl/artery#282] [Q&A]
Hello All,
I have been working on my V2I based master thesis and where I need to establish communication between car(vehicles) and RSU. Here as per the requirement, I have been able to invoke onWSM() from MyVeinsApp.cc, but I need onWSM() from my TraCIDemoRSU11p.cc to send message to cars and I should be able to receive message in onWSM() of MyVeinsApp.cc
My query here is: how to invoke message/data transfer betwen two .cc files (from TraCIDemoRSU11p.cc to MyVeinsApp.cc).

**Comment by riebl** [comment]
You should ask your question at the Veins project. Artery is not dealing with `onWSM` at all.
